src/ocean/io/serialize/StructSerializer.d

```diff
@@ -575,12 +575,8 @@ struct StructSerializer ( bool AllowUnions = false )
 
                 static if (isTypedef!(T))
                 {
-                    mixin(`
-                    static if ( is(T B == typedef) )
-                    {
-                        serializer.serialize(data, cast(B)(field), field_name);
-                    }
-                    `);
+                    serializer.serialize(data, cast(StripTypedef!(T))(field),
```

Comment:
So this is exactly why I don't want to allow silent `StripTypedef` application to D2 structs. Serializer must not reduce an entity to the base type but serialize it as actual struct. For current `Typedef` implementation ABI is compatible as deserialization will not show any immediate issues but it is an undefined behaviour time bomb.